go/vt/schemadiff/table_test.go

```diff
@@ -449,6 +452,88 @@ func TestCreateTableDiff(t *testing.T) {
 				"+	`y` int,",
 			},
 		},
+		{
+			name:  "added column with non deterministic expression, allow",
+			from:  "create table t1 (id int primary key, a int)",
+			to:    "create table t2 (id int primary key, a int, v varchar(36) not null default (uuid()))",
+			diff:  "alter table t1 add column v varchar(36) not null default (uuid())",
+			cdiff: "ALTER TABLE `t1` ADD COLUMN `v` varchar(36) NOT NULL DEFAULT (uuid())",
```

Comment:
If this always errors on MySQL, should it even be a flag on schemadiff or should we also always use error?

Reply:
Agreed. Removed the option altogether, the issue title is now misleading and I'll fix it. Confirmed that MySQL rejects this kind of statement even for empty tables (as opposed to [another issue we found](https://github.com/vitessio/vitess/pull/16678) the other day where the success depends on whether the table is empty or not). This is a bit of inconsistent behavior for MySQL, but at least we're consistent with it...